cs_tools/cli/tools/_scriptability/app.py

```diff
@@ -0,0 +1,363 @@
+# DEV NOTE:
+#
+# Future enhancements:
+#   - add tags on upload
+#   - add support for GUID mapping across instances of ThoughtSpot
+#   - ability to manipulate objects, such as renaming references to worksheets and tables, token replacement, etc.
+#   - add support for uploading .zip files
+#   - add support for tables and connections
+#
+
+import datetime
+import json
+import pathlib
+import click
+# import requests   # use httpx
+from thoughtspot_tml import YAMLTML
+from thoughtspot_tml.tml import TML
+from typer import Argument as A_, Option as O_  # noqa
+from typing import Dict, List, Optional
+import typer
+from zipfile import ZipFile
+
+from cs_tools import util
+from cs_tools.data.enums import GUID, TMLImportPolicy, TMLType, TMLContentType
+from cs_tools.api.middlewares import ConnectionMiddleware
+from cs_tools.cli.ux import _csv, console, CSToolsCommand, CSToolsGroup
+from cs_tools.cli.tools import common
+from cs_tools.cli.util import base64_to_file
+from cs_tools.thoughtspot import ThoughtSpot
+from cs_tools.cli.dependency import depends
+from cs_tools.cli.options import CONFIG_OPT, VERBOSE_OPT, TEMP_DIR_OPT
+
+
+def strip_blanks(inp: List[str]) -> List[str]:
+    return [e for e in inp if e]
+
+
+app = typer.Typer(
+    help="""
+    Tool for easily migrating TML between clusters.
+
+    ThoughtSpot provides the ability to extract object metadata (tables, worksheets, liveboards, etc.) 
+    in ThoughtSpot Modeling Language (TML) format, which is a text format based on YAML.  These files 
+    can be modified and uploaded into another instance. These files can then be modified and imported 
+    into another (or the same) instance to either create or modify objects.
+
+      cs_tools tools scriptability-migration --help
+    
+    \f
+    TODO - add more details on using the tools and a workflow.
+    
+    DEV NOTE:
+
+      Two control characters are offered in order to help with
+      docstrings in typer App helptext and command helptext
+      (docstrings).
+
+      \b - Preserve Whitespace / formatting.
+      \f - EOF, don't include anything after this character in helptext.
+    """,
+    cls=CSToolsGroup,
+    options_metavar='[--version, --help]'
+)
+
+
+@app.command(cls=CSToolsCommand)
+@depends(
+    thoughtspot=common.setup_thoughtspot,
+    options=[CONFIG_OPT, VERBOSE_OPT, TEMP_DIR_OPT],
+    enter_exit=True
+)
+def export(
+        ctx: click.Context,
+        tags: List[str] = O_([], metavar='TAGS',
+                             callback=_csv,
+                             help='list of tags to export for'),
+        export_ids: List[str] = O_([], metavar='GUIDS',
```

Comment:
if you wanted to name this `guids` instead of `export_ids` .. you're free to do that 😄 we don't have to follow the API naming conventions if we think it'll help users understand the CLI better

Reply:
I like following the API here.  It's a bit more typing, but think it's OK.